src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -243,6 +245,16 @@ public static void parseField(AbstractFieldMapper.Builder builder, String name,
                 parseCopyFields(propNode, builder);
             }
         }
+        if (indexAnalyzer != null) {
+            builder.indexAnalyzer(indexAnalyzer);
+        } else {
+            builder.indexAnalyzer(theAnalyzer);
+        }
+        if (searchAnalyzer != null) {
+            builder.searchAnalyzer(searchAnalyzer);
+        } else {
+            builder.searchAnalyzer(theAnalyzer);
```

Comment:
Strictly speaking this is different than the previous code - if none of index analyzer , search analyzer and theAnalyzer is set, we reset the indexAnalyzer and searchAnalyzer on the builder to null. We didn't use to do it...